src/test/regress/sql/aux_ao_rels_stat.sql

```diff
@@ -195,3 +195,174 @@ ORDER BY index;
 -- Drop AO and aux tables
 DROP TABLE ao_table;
 
+-- Test the fix: Adding data to a partitioned AO table using `COPY FROM`  
+-- does not change the `tupcount` in the segment table on the QD.
+CREATE OR REPLACE FUNCTION get_total_tupcount(table_name text)
+RETURNS bigint AS $$
+DECLARE
+    aoseg_table_name text;
+    tupcount_result bigint;
+BEGIN
+    -- Get the AO segment table name for the given partition
+    SELECT (n.nspname || '.' || c.relname) INTO aoseg_table_name
+    FROM pg_appendonly a
+        JOIN pg_class c ON c.oid = a.segrelid
+        JOIN pg_namespace n ON c.relnamespace = n.oid
+    WHERE a.relid = table_name::regclass::oid;
+
+    -- Sum tupcount from all segments
+    EXECUTE format('SELECT SUM(tupcount) FROM %s', aoseg_table_name)
+        INTO tupcount_result;
+
+    RETURN tupcount_result;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION get_tupcounts_for_partitions(main_table_name text)
+RETURNS TABLE (table_name text, tupcount bigint)
+AS $$
+BEGIN
+    -- Return main table tupcount first
+    RETURN QUERY
+    SELECT 
+        main_table_name::text AS table_name, 
+        get_total_tupcount(main_table_name) AS tupcount;
+
+    -- Return partition name and tupcount for each partition
+    RETURN QUERY 
+    SELECT 
+        p.partitiontablename::text, 
+        get_total_tupcount(p.partitiontablename)
+    FROM pg_partitions p
+    WHERE p.tablename = main_table_name
+    ORDER BY p.partitiontablename;
+END;
+$$ LANGUAGE plpgsql;
```

Comment:
```suggestion
CREATE OR REPLACE FUNCTION get_tupcounts_for_partitions(main_table_name text)
RETURNS TABLE (table_name text, tupcount bigint)
AS $$
BEGIN
    -- Return main table tupcount first
    RETURN QUERY
    SELECT
        main_table_name AS table_name,
        get_total_tupcount(main_table_name::regclass) AS tupcount;

    -- Return partition name and tupcount for each partition
    RETURN QUERY

    SELECT
        cl2.relname::text,
        get_total_tupcount(cl2.oid)
    FROM pg_partition pp
    JOIN pg_class cl1 ON pp.parrelid = cl1.oid
    JOIN pg_partition_rule pr ON pr.paroid = pp.oid
    JOIN pg_class cl2 ON cl2.oid = pr.parchildrelid
    WHERE cl1.oid = main_table_name::regclass
    ORDER BY 1;
END;
$$ LANGUAGE plpgsql;
```

Reply:
Can you explain the reason for these changes?  
For me, it seems clearer to use `table_name` as a parameter in `get_tupcounts_for_partitions`. What does this improve over the old code?

Reply:
And what's the purpose of returning the `table_schema` column in the test results?

Reply:
Your approach does not support schema, while mine - does
```sql
SELECT * FROM get_tupcounts_for_partitions('t_col_lev2');
 table_schema |          table_name           | tupcount 
--------------+-------------------------------+----------
 gpadmin      | t_col_lev2                    |         
 gpadmin      | t_col_lev2_1_prt_p1           |         
 gpadmin      | t_col_lev2_1_prt_p1_2_prt_sp1 |        2
 gpadmin      | t_col_lev2_1_prt_p1_2_prt_sp2 |        2
 gpadmin      | t_col_lev2_1_prt_p2           |         
 gpadmin      | t_col_lev2_1_prt_p2_2_prt_sp1 |        2
 gpadmin      | t_col_lev2_1_prt_p2_2_prt_sp2 |        2
(7 rows)

SELECT * FROM get_tupcounts_for_partitions('gpadmin.t_col_lev2');
 table_schema |          table_name           | tupcount 
--------------+-------------------------------+----------
 gpadmin      | t_col_lev2                    |         
 gpadmin      | t_col_lev2_1_prt_p1           |         
 gpadmin      | t_col_lev2_1_prt_p1_2_prt_sp1 |        2
 gpadmin      | t_col_lev2_1_prt_p1_2_prt_sp2 |        2
 gpadmin      | t_col_lev2_1_prt_p2           |         
 gpadmin      | t_col_lev2_1_prt_p2_2_prt_sp1 |        2
 gpadmin      | t_col_lev2_1_prt_p2_2_prt_sp2 |        2
(7 rows)
```

Reply:
I still don't understand why you suggest complicating the code.  
- These functions are written for specific tests and rely on the current default search path.  
- The tables are created in the current file with the currently set search path.

So, even if someone changes the schema beforehand, the old code should still work. 

```sql
postgres=# CREATE SCHEMA my_new_schema;
CREATE SCHEMA
postgres=# SET search_path TO my_new_schema;
SET
-- creating function
-- create table t
-- insert data into t
postgres=# SELECT * FROM get_tupcounts_for_partitions('t');
 table_name | tupcount 
------------+----------
 t          |         
 t_1_prt_1  |       50
 t_1_prt_1  |       50
 t_1_prt_2  |       50
 t_1_prt_2  |       50
(5 rows)
```
But if you insist, I can add this change. Should I do it?

Reply:
I still don't understand why you suggest complicating the code.  
- These functions are written for specific tests and rely on the current default search path.  
- The tables are created in the current file with the currently set search path.

So, even if someone changes the schema beforehand, the old code should still work. 

```sql
postgres=# CREATE SCHEMA my_new_schema;
CREATE SCHEMA
postgres=# SET search_path TO my_new_schema;
SET
-- creating function
-- create table t
-- insert data into t
postgres=# SELECT * FROM get_tupcounts_for_partitions('t');
 table_name | tupcount 
------------+----------
 t          |         
 t_1_prt_1  |       50
 t_1_prt_1  |       50
 t_1_prt_2  |       50
 t_1_prt_2  |       50
(5 rows)
```

Reply:
Haha))) it dawned on me that it would break if someone put a table with the same name into another scheme.
Thanks for help

Reply:
I've changed `get_total_tupcount()` to work directly with OID but kept the part that constructs the name as `n.nspname || '.' || c.relname`. Additionally, I added a condition in `get_tupcounts_for_partitions()` to ensure it searches only within the current schema.

Reply:
```sql
SELECT * FROM get_tupcounts_for_partitions('t_col_lev2');
          table_name           | tupcount 
-------------------------------+----------
 t_col_lev2                    |         
 t_col_lev2_1_prt_p1           |         
 t_col_lev2_1_prt_p1_2_prt_sp1 |        2
 t_col_lev2_1_prt_p1_2_prt_sp2 |        2
 t_col_lev2_1_prt_p2           |         
 t_col_lev2_1_prt_p2_2_prt_sp1 |        2
 t_col_lev2_1_prt_p2_2_prt_sp2 |        2
(7 rows)

SELECT * FROM get_tupcounts_for_partitions('gpadmin.t_col_lev2');
     table_name     | tupcount 
--------------------+----------
 gpadmin.t_col_lev2 |         
(1 row)

```

Reply:
Ok, I added schema support as you showed. This functionality can now be used in other places.